### PR TITLE
Add CITATION.cff

### LINKED ISSUE #207: Update CITATION
Add @tmigot and @amontoison and use CITATION.cff (https://citation-file-format.github.io/cff-initializer-javascript/#/).

### PR DESCRIPTION
**Pull request by tmigot**
close #207 

The bibtex generated by github is:

@software{Migot_ADNLPModels_jl_Automatic_Differentiation,
author = {Migot, Tangi and Montoison, Alexis and Orban, Dominique and Soares Siqueira, Abel and contributors},
license = {MPL-2.0},
title = {{ADNLPModels.jl: Automatic Differentiation models implementing the NLPModels API}},
url = {https://github.com/JuliaSmoothOptimizers/ADNLPModels.jl}
}

### CODE REVIEW

**Comment by amontoison** [comment]
I just have a minor comment.
Otherwise, it looks good to me.

File: LICENSE.md [hunk 1]
```diff
@@ -1,4 +1,4 @@
-Copyright (c) 2015-2021: Dominique Orban and Abel Soares Siqueira
+Copyright (c) 2015-2024: Tangi Migot, Alexis Montoison, Dominique Orban and Abel Soares Siqueira
```
[the remark this answers is not in the document]
Comment (tmigot) [reply LICENSE.md]: Good idea, thanks

**Review by amontoison — APPROVED** [review]

**Review by dpo — COMMENTED** [review]
That’s great, but what BibTeX style recognizes `@software`???

**Comment by abelsiqueira** [comment]
> That’s great, but what BibTeX style recognizes `@software`???

Here's the relevant discussion on why GitHub parses the CFF into `@software`: https://github.com/citation-file-format/ruby-cff/issues/40

**Comment by dpo** [comment]
I don't know a *single* journal that uses BibLaTeX. Some still require explicit `\bibitem{...}`. In fact, most still use LaTeX (not pdfLaTeX).

**Comment by dpo** [comment]
I just saw https://github.com/citation-file-format/ruby-cff/issues/40#issuecomment-903684216. That's encouraging. Could you please check that the SIAM and Springer styles behave that way?

**Comment by abelsiqueira** [comment]
I don't think it matters whether it works for some style, these are two different things. CFF has the metadata for software citation that is useful for Zenodo and possibly other metadata ingesting tool. In particular, the cite me button on GitHub translates to bibtex, which people might want to use to cite. If that doesn't work because it's `@software`, it might fall back to `@misc`. If that doesn't work, they'll go back to the DOI and cite that. If we have a paper related to this software, we can append that to CITATION.cff with the field `preferred-citation`, and the file will keep working for all main purposes.

**Comment by tmigot** [comment]
I just checked on Overleaf with a couple of projects using Springer `\documentclass[smallextended]{svjour3}`  and it does uses Misc automatically it seems.

**Comment by dpo** [comment]
It matters. We’re citing packages all the time in our papers. So if common styles, such as Springer or SIAM didn’t support `@software`, it would become a real drag. But fortunately, it appears that that is not the case.

**Pull request merged**